src/main/java/org/opensearch/flowframework/transport/DeprovisionWorkflowTransportAction.java

```diff
@@ -226,15 +226,23 @@ private void updateWorkflowState(
     ) {
         if (remainingResources.isEmpty()) {
             // Successful deprovision, reset state to initial
-            flowFrameworkIndicesHandler.putInitialStateToWorkflowState(
-                workflowId,
-                getUserContext(client),
-                ActionListener.wrap(indexResponse -> {
-                    logger.info("Reset workflow {} state to NOT_STARTED", workflowId);
-                }, exception -> { logger.error("Failed to reset to initial workflow state for {}", workflowId, exception); })
-            );
-            // return workflow ID
-            listener.onResponse(new WorkflowResponse(workflowId));
+            flowFrameworkIndicesHandler.doesTemplateExist(workflowId, templateExists -> {
+                if (Boolean.TRUE.equals(templateExists)) {
+                    flowFrameworkIndicesHandler.putInitialStateToWorkflowState(
+                        workflowId,
+                        getUserContext(client),
+                        ActionListener.wrap(indexResponse -> {
+                            logger.info("Reset workflow {} state to NOT_STARTED", workflowId);
+                        }, exception -> { logger.error("Failed to reset to initial workflow state for {}", workflowId, exception); })
+                    );
+                } else {
+                    flowFrameworkIndicesHandler.deleteFlowFrameworkSystemIndexDoc(workflowId, ActionListener.wrap(deleteResponse -> {
+                        logger.info("Deleted workflow {} state", workflowId);
+                    }, exception -> { logger.error("Failed to delete workflow state for {}", workflowId, exception); }));
```

Comment:
should we also have   `listener.onFailure(new FlowFrameworkException(errorMessage, ExceptionsHelper.status(exception)));` here?

Reply:
I kept this change consistent with how it was previously.   If the deprovision was successful the listener has already been completed with `onResponse()` so `onFailure()` here would be a no-op.

I'm open to discussing whether we want to delay the `onResponse()` for a successful deprovisioning until after we write the state index, but would like that discussion for 2.15.

Reply:
Yeah this can be done later too, as we do it like this now

Reply:
https://github.com/opensearch-project/flow-framework/issues/691